src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs

```diff
@@ -267,6 +268,15 @@ private async Task UseCallback_BadCertificate_ExpectedPolicyErrors_Helper(string
 
                 handler.ServerCertificateCustomValidationCallback = (request, cert, chain, errors) =>
                 {
+                    // https://github.com/dotnet/corefx/issues/21922#issuecomment-315555237
+                    X509ChainStatusFlags flags = chain.ChainStatus.Aggregate(X509ChainStatusFlags.NoError, (cur, status) => cur | status.Status);
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) &&
+                        flags == X509ChainStatusFlags.RevocationStatusUnknown &&
```

Comment:
If RevocationStatusUnknown is causing failure on macOS, but not on Windows or Linux, then that's a bug in the cert layer.

If it's just that macOS comes back with RevocationStatusUnknown and the others don't... that's not a thing we can really fix.  But, you/we could decide that SslStream's default chain policy should say that RevocationStatusUnknown should not be considered as failure.  (That'd be ChainPolicy.RevocationFlags)

Reply:
Oh I have mixed feelings, but that needs to be a different and wider discussion than just changing tests.

Reply:
> This is just changing the test to say that if you see RevocationStatusUnknown you want to expect failure

It's more like "on this known faulty server we expect this specific SslPolicyError failure". Removing the `CheckCertificateRevocationList` from the condition is fine, but I would like to keep the OSX platform check to be explicit that the workaround is for that platform only.